src/main.js

```diff
@@ -284,21 +285,23 @@ function handleClose(streams, children, childrenInfo) {
         if (aliveChildren.length === 0) {
             exit(exitCodes);
         }
-    });
 
-    if (config.killOthers) {
```

Comment:
The `--kill-others` flag states "kill other processes if one exits or dies".
So testing the exit code should probably not be done, otherwise we may cause more breakage.